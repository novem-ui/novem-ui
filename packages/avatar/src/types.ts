export type AvatarSizes = 'sm' | 'md' | 'lg'

export interface AvatarProps {
  description?: string
  image?: string
  name?: string
  size?: AvatarSizes
}
