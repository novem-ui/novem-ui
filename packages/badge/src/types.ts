import theme from '@novem-ui/theme'
import { Hierarchy } from '@novem-ui/base'
import { Theme } from '@emotion/react'

export type BadgeHierarchy = Exclude<Hierarchy, 'link'>
export type BadgeSize = 'sm' | 'lg'

export interface BadgeProps {
  hierarchy?: BadgeHierarchy
  size?: BadgeSize
  baseColor?: keyof Omit<Omit<typeof theme.colors.palette, 'base'>, 'darkGrey'>
}

export type BadgeUtilsFunction = (props: BadgeProps & { theme?: Theme }) => string
