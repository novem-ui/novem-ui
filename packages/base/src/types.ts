import { Property } from 'csstype'
import { BaseColor, Color, SpacingValue } from '@novem-ui/theme'
import { Theme } from '@emotion/react'

// Spacing types
export type SpacingPropName =
  | 'm'
  | 'mt'
  | 'mr'
  | 'mb'
  | 'ml'
  | 'mx'
  | 'my'
  | 'margin'
  | 'marginX'
  | 'marginY'
  | 'marginTop'
  | 'marginRight'
  | 'marginBottom'
  | 'marginLeft'
  | 'p'
  | 'pt'
  | 'pr'
  | 'pb'
  | 'pl'
  | 'px'
  | 'py'
  | 'padding'
  | 'paddingX'
  | 'paddingY'
  | 'paddingTop'
  | 'paddingRight'
  | 'paddingBottom'
  | 'paddingLeft'

export type SpacingPropValue = SpacingValue | string

export type SpacingPropEntry = [SpacingPropName, SpacingPropValue]

export type SpacingProps = { [key in SpacingPropName]?: SpacingPropValue }

export type CSSSpacingProps = Omit<
  SpacingProps,
  'm' | 'mt' | 'mr' | 'mb' | 'ml' | 'mx' | 'my' | 'p' | 'pt' | 'pr' | 'pb' | 'pl' | 'px' | 'py'
>

// Text types
export interface TypographyProps<As, Size> {
  align?: Property.TextAlign
  as?: As
  color?: Color | Property.Color
  decoration?: Property.TextDecoration
  direction?: Property.Direction
  family?: Property.FontFamily
  kerning?: Property.FontKerning
  letterSpacing?: Property.LetterSpacing
  lineHeight?: Property.LineHeight
  orientation?: Property.TextOrientation
  transform?: Property.TextTransform
  size?: Size | Property.FontSize
  shadow?: Property.TextShadow
  stretch?: Property.FontStretch
  fontStyle?: Property.FontStyle
  theme?: Theme
  variantCaps?: Property.FontVariantCaps
  variantNumeric?: Property.FontVariantNumeric
  weight?: Property.FontWeight
  whiteSpace?: Property.FontWeight
  wordBreak?: Property.WordBreak
  wordSpacing?: Property.WordSpacing
  writingMode?: Property.WritingMode
}

export type TextComponentType = 'heading' | 'text'
export type HeadingTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

// Hierarchy
export type Hierarchy = 'loud' | 'quiet' | 'link'

// InputElement
export interface InputElementProps {
  theme?: Theme
  baseColor?: BaseColor
  error?: boolean | string | Error
  valid?: boolean
  disabled?: boolean
  readOnly?: boolean
}

// Layout

export interface LayoutProps {
  display?: Property.Display
  position?: Property.Position
  top?: Property.Top
  bottom?: Property.Bottom
  height?: Property.Height
  left?: Property.Left
  right?: Property.Right
  width?: Property.Width
}
