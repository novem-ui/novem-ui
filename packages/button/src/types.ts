import { Hierarchy, SpacingProps } from '@novem-ui/base'
import theme from '@novem-ui/theme'

export type ButtonSize = 'sm' | 'md' | 'lg'

export interface ButtonProps extends SpacingProps{
  hierarchy?: Hierarchy
  size?: ButtonSize
  baseColor?: keyof typeof theme.colors.palette
  disabled?: boolean
}
