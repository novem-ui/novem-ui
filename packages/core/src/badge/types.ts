import * as React from 'react'

export default interface IBadge {
  variant: string,
  children: React.ReactElement,
}
