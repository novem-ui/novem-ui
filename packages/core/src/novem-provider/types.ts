import { Theme } from "@novem-ui/theme";

export interface NovemProviderProps {
  theme?: Theme;
  reset?: boolean;
  baseColor?: string;
}
