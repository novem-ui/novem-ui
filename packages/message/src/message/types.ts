import theme from "@novem-ui/theme";

export interface ButtonActionTypes {
  text: string
  action: () => void
}

export interface MessageTypes {
  title?: string
  message: string
  onClose?: () => void
  button?: ButtonActionTypes
  baseColor?: keyof typeof theme.colors.palette
}
