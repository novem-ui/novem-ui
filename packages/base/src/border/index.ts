import { css } from '@emotion/react'
import { Theme } from '@novem-ui/theme'

import { BorderProps } from '../types'
import getColorFactory from '../utils/get-color'

const border = ({
  border,
  borderBlock,
  borderBlockColor,
  borderBlockEnd,
  borderBlockEndColor,
  borderBlockEndStyle,
  borderBlockEndWidth,
  borderBlockStart,
  borderBlockStartColor,
  borderBlockStartStyle,
  borderBlockStartWidth,
  borderBlockStyle,
  borderBlockWidth,
  borderBottom,
  borderBottomColor,
  borderBottomLeftRadius,
  borderBottomRightRadius,
  borderBottomStyle,
  borderBottomWidth,
  borderCollapse,
  borderColor,
  borderEndEndRadius,
  borderEndStartRadius,
  borderInline,
  borderInlineColor,
  borderInlineEnd,
  borderInlineEndColor,
  borderInlineEndStyle,
  borderInlineEndWidth,
  borderInlineStart,
  borderInlineStartColor,
  borderInlineStartStyle,
  borderInlineStartWidth,
  borderInlineStyle,
  borderInlineWidth,
  borderLeft,
  borderLeftColor,
  borderLeftStyle,
  borderLeftWidth,
  borderRadius,
  borderRight,
  borderRightColor,
  borderRightStyle,
  borderRightWidth,
  borderSpacing,
  borderStartEndRadius,
  borderStartStartRadius,
  borderStyle,
  borderTop,
  borderTopColor,
  borderTopLeftRadius,
  borderTopRightRadius,
  borderTopStyle,
  borderTopWidth,
  borderWidth,
  theme
}: BorderProps & { theme: Theme }) => {
  const getColor = getColorFactory(theme)
  
  return css`
    ${border && `border: ${border};`}
    ${borderBlock && `border-block: ${borderBlock};`}
    ${borderBlockColor && `border-block-color: ${getColor(borderBlockColor)};`}
    ${borderBlockEnd && `border-block-end: ${borderBlockEnd};`}
    ${borderBlockEndColor && `border-block-end-color: ${getColor(borderBlockEndColor)};`}
    ${borderBlockEndStyle && `border-block-end-style: ${borderBlockEndStyle};`}
    ${borderBlockEndWidth && `border-block-end-width: ${borderBlockEndWidth};`}
    ${borderBlockStart && `border-block-start: ${borderBlockStart};`}
    ${borderBlockStartColor && `border-block-start-color: ${getColor(borderBlockStartColor)};`}
    ${borderBlockStartStyle && `border-block-start-style: ${borderBlockStartStyle};`}
    ${borderBlockStartWidth && `border-block-start-width: ${borderBlockStartWidth};`}
    ${borderBlockStyle && `border-block-style: ${borderBlockStyle};`}
    ${borderBlockWidth && `border-block-width: ${borderBlockWidth};`}
    ${borderBottom && `border-bottom: ${borderBottom};`}
    ${borderBottomColor && `border-bottom-color: ${getColor(borderBottomColor)};`}
    ${borderBottomLeftRadius && `border-bottom-left-radius: ${borderBottomLeftRadius};`}
    ${borderBottomRightRadius && `border-bottom-right-radius: ${borderBottomRightRadius};`}
    ${borderBottomStyle && `border-bottom-style: ${borderBottomStyle};`}
    ${borderBottomWidth && `border-bottom-width: ${borderBottomWidth};`}
    ${borderCollapse && `border-collapse: ${borderCollapse};`}
    ${borderColor && `border-color: ${getColor(borderColor)};`}
    ${borderEndEndRadius && `border-end-end-radius: ${borderEndEndRadius};`}
    ${borderEndStartRadius && `border-end-start-radius: ${borderEndStartRadius};`}
    ${borderInline && `border-inline: ${borderInline};`}
    ${borderInlineColor && `border-inline-color: ${getColor(borderInlineColor)};`}
    ${borderInlineEnd && `border-inline-end: ${borderInlineEnd};`}
    ${borderInlineEndColor && `border-inline-end-color: ${getColor(borderInlineEndColor)};`}
    ${borderInlineEndStyle && `border-inline-end-style: ${borderInlineEndStyle};`}
    ${borderInlineEndWidth && `border-inline-end-width: ${borderInlineEndWidth};`}
    ${borderInlineStart && `border-inline-start: ${borderInlineStart};`}
    ${borderInlineStartColor && `border-inline-start-color: ${getColor(borderInlineStartColor)};`}
    ${borderInlineStartStyle && `border-inline-start-style: ${borderInlineStartStyle};`}
    ${borderInlineStartWidth && `border-inline-start-width: ${borderInlineStartWidth};`}
    ${borderInlineStyle && `border-inline-style: ${borderInlineStyle};`}
    ${borderInlineWidth && `border-inline-width: ${borderInlineWidth};`}
    ${borderLeft && `border-left: ${borderLeft};`}
    ${borderLeftColor && `border-left-color: ${getColor(borderLeftColor)};`}
    ${borderLeftStyle && `border-left-style: ${borderLeftStyle};`}
    ${borderLeftWidth && `border-left-width: ${borderLeftWidth};`}
    ${borderRadius && `border-radius: ${borderRadius};`}
    ${borderRight && `border-right: ${borderRight};`}
    ${borderRightColor && `border-right-color: ${getColor(borderRightColor)};`}
    ${borderRightStyle && `border-right-style: ${borderRightStyle};`}
    ${borderRightWidth && `border-right-width: ${borderRightWidth};`}
    ${borderSpacing && `border-spacing: ${borderSpacing};`}
    ${borderStartEndRadius && `border-start-end-radius: ${borderStartEndRadius};`}
    ${borderStartStartRadius && `border-start-start-radius: ${borderStartStartRadius};`}
    ${borderStyle && `border-style: ${borderStyle};`}
    ${borderTop && `border-top: ${borderTop};`}
    ${borderTopColor && `border-top-color: ${getColor(borderTopColor)};`}
    ${borderTopLeftRadius && `border-top-left-radius: ${borderTopLeftRadius};`}
    ${borderTopRightRadius && `border-top-right-radius: ${borderTopRightRadius};`}
    ${borderTopStyle && `border-top-style: ${borderTopStyle};`}
    ${borderTopWidth && `border-top-width: ${borderTopWidth};`}
    ${borderWidth && `border-width: ${borderWidth};`}
  `
}

export default border
