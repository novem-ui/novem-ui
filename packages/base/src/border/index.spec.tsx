import React from 'react'

import { render } from '@testing-library/react'
import styled from '@emotion/styled'
import theme from '@novem-ui/theme'
import { matchers } from '@emotion/jest'

import { BorderProps } from '../types'

import border from '.'

beforeAll(() => {
  expect.extend(matchers)
})

const ComponentWithBorder = styled.div<BorderProps>`
  ${border}
`

describe('borderColor', () => {
  test('should return a component with the correct border color and 1px width', () => {
    const { container } = render(<ComponentWithBorder theme={theme} border="1px solid" borderColor="blue.500" />)

    expect(container).toMatchSnapshot()
  })

  test('should return a component with unmapped colors', () => {
    const { container } = render(
      <ComponentWithBorder theme={theme} borderColor="blu.e.500" borderBlockColor="violet" />
    )

    expect(container).toMatchSnapshot()
  })

  test('should return a component with all the props', () => {
    const { container } = render(
      <ComponentWithBorder
        theme={theme}
        border="1px solid"
        borderBlock="AppWorkspace"
        borderBlockColor="green.700"
        borderBlockEnd="unset"
        borderBlockEndColor="yellow.200"
        borderBlockEndStyle="outset"
        borderBlockEndWidth="thick"
        borderBlockStart="1px solid"
        borderBlockStartColor="indigo.700"
        borderBlockStartStyle="revert"
        borderBlockStartWidth="1px solid"
        borderBlockStyle="double"
        borderBlockWidth="1px solid"
        borderBottom="1px solid"
        borderBottomColor="indigo.700"
        borderBottomLeftRadius="1px solid"
        borderBottomRightRadius="1px solid"
        borderBottomStyle="hidden"
        borderBottomWidth="1px solid"
        borderCollapse="collapse"
        borderColor="indigo.700"
        borderEndEndRadius="inherit"
        borderEndStartRadius="unset"
        borderInline="1px solid"
        borderInlineColor="indigo.700"
        borderInlineEnd="antiquewhite"
        borderInlineEndColor="indigo.700"
        borderInlineEndStyle="initial"
        borderInlineEndWidth="initial"
        borderInlineStart="initial"
        borderInlineStartColor="indigo.700"
        borderInlineStartStyle="outset"
        borderInlineStartWidth="outset"
        borderInlineStyle="outset"
        borderInlineWidth="outset"
        borderLeft="outset"
        borderLeftColor="indigo.700"
        borderLeftStyle="outset"
        borderLeftWidth="outset"
        borderRadius="outset"
        borderRight="outset"
        borderRightColor="indigo.700"
        borderRightStyle="ridge"
        borderRightWidth="thick"
        borderSpacing="unset"
        borderStartEndRadius="1px"
        borderStartStartRadius="1px"
        borderStyle="hidden"
        borderTop="hidden"
        borderTopColor="indigo.700"
        borderTopLeftRadius="inherit"
        borderTopRightRadius="inherit"
        borderTopStyle="ridge"
        borderTopWidth="thin"
        borderWidth="inherit"
      />
    )

    expect(container).toMatchSnapshot()
  })
})
