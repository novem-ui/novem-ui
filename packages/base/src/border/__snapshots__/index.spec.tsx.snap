// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`borderColor should return a component with all the props 1`] = `
.emotion-0 {
  border: 1px solid;
  border-block: AppWorkspace;
  border-block-color: #378949;
  border-block-end: unset;
  border-block-end-color: #F1694B;
  border-block-end-style: outset;
  border-block-end-width: thick;
  border-block-start: 1px solid;
  border-block-start-color: #2c2489;
  border-block-start-style: revert;
  border-block-start-width: 1px solid;
  border-block-style: double;
  border-block-width: 1px solid;
  border-bottom: 1px solid;
  border-bottom-color: #2c2489;
  border-bottom-left-radius: 1px solid;
  border-bottom-right-radius: 1px solid;
  border-bottom-style: hidden;
  border-bottom-width: 1px solid;
  border-collapse: collapse;
  border-color: #2c2489;
  border-end-end-radius: inherit;
  border-end-start-radius: unset;
  border-inline: 1px solid;
  border-inline-color: #2c2489;
  border-inline-end: antiquewhite;
  border-inline-end-color: #2c2489;
  border-inline-end-style: initial;
  border-inline-end-width: initial;
  border-inline-start: initial;
  border-inline-start-color: #2c2489;
  border-inline-start-style: outset;
  border-inline-start-width: outset;
  border-inline-style: outset;
  border-inline-width: outset;
  border-left: outset;
  border-left-color: #2c2489;
  border-left-style: outset;
  border-left-width: outset;
  border-radius: outset;
  border-right: outset;
  border-right-color: #2c2489;
  border-right-style: ridge;
  border-right-width: thick;
  border-spacing: unset;
  border-start-end-radius: 1px;
  border-start-start-radius: 1px;
  border-style: hidden;
  border-top: hidden;
  border-top-color: #2c2489;
  border-top-left-radius: inherit;
  border-top-right-radius: inherit;
  border-top-style: ridge;
  border-top-width: thin;
  border-width: inherit;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`borderColor should return a component with the correct border color and 1px width 1`] = `
.emotion-0 {
  border: 1px solid;
  border-color: #3564e3;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`borderColor should return a component with unmapped colors 1`] = `
.emotion-0 {
  border-block-color: violet;
  border-color: blu.e.500;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
